packages/gatsby/src/utils/client-assets-for-template.ts

```diff
@@ -63,13 +63,21 @@ export async function getScriptsAndStylesForTemplate(
     }
   }
 
+  // console.log(componentChunkName)
+
   // Pick up scripts and styles that are used by a template using webpack.stats.json
   for (const chunkName of [`app`, componentChunkName]) {
-    const assets = webpackStats.assetsByChunkName[chunkName]
+    let assets = webpackStats.assetsByChunkName[chunkName]
+
     if (!assets) {
       continue
     }
 
+    // Remove JS asset for templates
+    if (chunkName !== `app`) {
+      assets = assets.filter(asset => !asset.endsWith(`.js`))
+    }
```

Comment:
This removes "top level" assets, but below we also have handling for "webpack magic comments" and we should filter out `.js` out as well there I think (just for page template ones, not for the `app`)?

Reply:
Oh yes! Good catch